providers/google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/GoogleComputeEngineServiceAdapter.java

```diff
@@ -142,9 +146,18 @@
       // Add tags for firewalls
       FirewallTagNamingConvention naming = firewallTagNamingConvention.get(group);
       List<String> ports = simplifyPorts(options.getInboundPorts());
-      if (ports != null){
+      if (ports != null) {
          tags.add(naming.name(ports));
       }
+      // Add tags coming from firewalls (aka securityGroups)
+      if (!options.getGroups().isEmpty()) {
+         for (String firewallName : options.getGroups()) {
+            Firewall firewall = api.firewalls().get(firewallName);
+            validateFirewall(firewall, network);
```

Comment:
This is redundant, as it has already been validated in the `CreateNodesWithGroupEncodedIntoNameThenAddToSet`.